### PR TITLE
Edit Readme (and fix bugs)

### PR DESCRIPTION
**Pull request by laulauland**
This PR moves significantly reduces the length of the Readme by moving most of the documentation to the docs.autometrics.dev site ([draft PR](https://github.com/autometrics-dev/docs/pull/10))

It also patches a small issue that would sometimes crash the TypeScript plugin.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,105 +2,99 @@
...
+<a href="https://github.com/autometrics-dev/autometrics-ts/actions?query=branch%3Amain"><img src="https://github.com/autometrics-dev/autometrics-ts/actions/workflows/ci.yml/badge.svg?event=push&branch=main" alt="Autometrics CI status" /></a>
 <a href="https://opensource.org/licenses/MIT" rel="nofollow"><img src="https://img.shields.io/npm/l/@autometrics/autometrics" alt="License"></a>
 <a href="https://discord.gg/MJr7pYzZQ4" rel="nofollow"><img src="https://img.shields.io/discord/950489382626951178?label=Discord&logo=discord&logoColor=white" alt="discord server"></a>
 </div>
 
 <hr />
 
-> A TypeScript port of the Rust
-> [autometrics-rs](https://github.com/fiberplane/autometrics-rs) library
+Autometrics library provides a wrapper for functions and decorator
```
Comment (emschwartz) [inline README.md:12]: I'd also include the problem statement part

```
Metrics are a powerful and cost-efficient tool for understanding the health and performance of your code in production. But it's hard to decide what metrics to track and even harder to write queries to understand the data.
```

Then, I'd rephrase this to say
```
Autometrics provides a wrapper function and decorator to instrument functions, classes, and methods with the most useful metrics: request rate, error rate, and latency. It standardizes these metrics and then generates powerful Prometheus queries based on your function details to help you quickly identify and debug issues in production.
```

File: README.md [hunk 2]
```diff
@@ -2,105 +2,99 @@
...
+for classes and methods to instrument them with the most useful
+metrics: request rate, error rate, and latency. It then uses the instrumented
+function names to generate Prometheus queries so you don’t need to hand-write
+complicated PromQL.
 
-**A library that makes it easy to understand the error rate, response time, and
-production usage of any function in your code.** Jump straight from your IDE to
-live Prometheus charts for each HTTP/RPC handler, database method, or other
-piece of application logic.
+[Learn more about Autometrics here](https://autometrics.dev/).
```
Comment (emschwartz) [inline README.md:18]:
```suggestion
[Learn more about Autometrics at autometrics.dev](https://autometrics.dev/).
```

File: README.md [hunk 3]
```diff
@@ -2,105 +2,99 @@
...
-See [Why Autometrics?](https://github.com/autometrics-dev#why-autometrics) for
-more details on the ideas behind autometrics
-
-![AutometricsTS demo](./assets/autometrics-ts-demo.gif)
 
 ## Features
 
-- ✨ `autometrics` wrapper instruments any function or class method to track its
+- ✨ `autometrics()` wrapper / `@Autometrics()` instruments any function or
+class method to track its
```
Comment (emschwartz) [inline README.md:24]:
```suggestion
class method to track the
```

Comment (emschwartz) [reply README.md:24]: Also, I think it's better to not manually wrap lines and just let editors do it

File: README.md [hunk 4]
```diff
@@ -2,105 +2,99 @@
...
-
-See [Why Autometrics?](https://github.com/autometrics-dev#why-autometrics) for
-more details on the ideas behind autometrics
-
-![AutometricsTS demo](./assets/autometrics-ts-demo.gif)
 
 ## Features
 
-- ✨ `autometrics` wrapper instruments any function or class method to track its
+- ✨ `autometrics()` wrapper / `@Autometrics()` instruments any function or
```
Comment (emschwartz) [inline README.md:23]: It's a little confusing to me what `@Autometrics()` is

File: README.md [hunk 5]
```diff
@@ -2,105 +2,99 @@
...
+class method to track its
   most useful metrics
-- 🌳 Works in NodeJS and browser environments (Deno and serverless support coming soon)
+- 🌳 Works in NodeJS, experimental support for browser and Deno environments
```
Comment (emschwartz) [inline README.md:26]:
```suggestion
- 🌳 Works in NodeJS and has experimental support for Deno and browser environments
```

File: README.md [hunk 6]
```diff
@@ -2,105 +2,99 @@
...
 - 💡 Writes Prometheus queries so you can understand the data generated without
   knowing PromQL
 - 🔗 Injects links to live Prometheus charts directly into each function's doc
+- 🔍 Helps you to [identify commits](https://docs.autometrics.dev/typescript/adding-version-information) that introduced errors or increased latency
+- 🚨 Allows you to [define alerts](https://docs.autometrics.dev/typescript/adding-alerts-and-slos) using SLO best practices directly in your source code
```
Comment (emschwartz) [inline README.md:31]: Similar to the Rust readme redo, I would separate the benefits from the advanced features (and name the first section Benefits)

File: README.md [hunk 7]
```diff
@@ -2,105 +2,99 @@
...
+```bash
 npm install --save autometrics
+# or
+yarn add --save autometrics
+# or
+pnpm add --save autometrics
 ```
 
-Use the library in your code:
+2. Instrument your code using the `autometrics` wrapper or `Autometrics`
```
Comment (emschwartz) [inline README.md:65]: It might be worth showing a tiny example of using the decorator on a minimal class

File: README.md [hunk 8]
```diff
@@ -2,105 +2,99 @@
...
 
-Import and use the library in your code:
+3. Configure Prometheus to scrape the data
 
-```typescript
-import { autometrics } from "@autometrics/autometrics"
-```
+By default the TypeScript library makes the metrics available on
+`<your_host>:9464/metrics`. Make sure your Prometheus is configured correctly to
+find it.
```
Comment (emschwartz) [inline README.md:82]:
```suggestion
scrape it.
```

File: README.md [hunk 9]
```diff
@@ -2,105 +2,99 @@
...
 ```
 
-Import and use the library in your code:
+3. Configure Prometheus to scrape the data
 
-```typescript
-import { autometrics } from "@autometrics/autometrics"
-```
+By default the TypeScript library makes the metrics available on
+`<your_host>:9464/metrics`. Make sure your Prometheus is configured correctly to
```
Comment (emschwartz) [inline README.md:81]: Instead of saying "make sure", just tell people how to do this

File: README.md [hunk 10]
```diff
@@ -2,105 +2,99 @@
...
+find it.
+
+[See the docs](https://docs.autometrics.dev/configuring-prometheus/local) for
+example Prometheus configuration.
 
-## Getting PromQL queries
+4. Install the IDE extension
 
-In order to get PromQL query links in your IDE download the [Autometrics VSCode
+In order to get PromQL query links in your VSCode, download the [Autometrics VSCode
```
Comment (emschwartz) [inline README.md:89]: It's actual charts now, not just links

Comment (flenter) [reply README.md:89]: I would leave out the "your"

**Review by flenter — APPROVED** [review]
🎉
